### DISCUSSION TITLE
Where to get chanel id from channel link

### DISCUSSION THREAD

**Discussion by DinhHuy2010** [ThioJoe/YT-Spammer-Purge#614] [Q&A] [answered]
Where is the code to get channel id from channel link?

**Comment by TechStudent10** [comment]
I have this channel URL (https://www.youtube.com/channel/UCzb9_b2UY29xuY-S8BsmpOg). The part after `/channel/` is the channel ID.

However, you should be able to paste the link in just like that.

**Reply by ThioJoe** [reply]
Also, one sure way to get the link with the channel ID in it (as opposed to username), is to go to the watch page for any video uploaded by a channel, then click on their channel name from there, and it will have the /channel/ link version in the URL bar.

**Reply by TechStudent10** [reply]
yeah I noticed that was a problem when I searched up any channel that has a custom channel URL (which I don't) so I used my channel as an example.

**Reply by DinhHuy2010** [reply]
not that, the code to get channel id from the channel link without do that

**Reply by ThioJoe** [reply]
If you're asking about the actual code in the program, it's this function: https://github.com/ThioJoe/YT-Spammer-Purge/blob/d5d943a3f9752c00dd190536453db04d0200ff7e/Scripts/validation.py#L102

Just note it's not perfect, and sometimes returns the wrong channel if a /c/ type URL is given, because of the nature of how YouTube's API requires you to fetch it

**Reply by DinhHuy2010** [reply]
I guess I have the code (You can test it)
https://github.com/ThioJoe/YT-Spammer-Purge/discussions/655

**Comment by TechStudent10** [comment, marked as answer]
Copied and Pasted from the previous thread by ThioJoe;

---

If you're asking about the actual code in the program, it's this function:

https://github.com/ThioJoe/YT-Spammer-Purge/blob/d5d943a3f9752c00dd190536453db04d0200ff7e/Scripts/validation.py#L102

Just note it's not perfect, and sometimes returns the wrong channel if a /c/ type URL is given, because of the nature of how YouTube's API requires you to fetch it
